api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1381,7 +1382,61 @@ public void getObject(String bucketName, String objectName, String fileName)
     }
   }
 
-
+  /**
+   * Copy a source object into a new object with the provided name in the provided bucket.
+   *
+   * </p>
+   * <b>Example:</b><br>
+   * 
+   * <pre>
+   * {@code minioClient.copyObject("my-bucketname", "my-objectname", "/my-sourcebucketname/my-objectsourcename", 
+   * copyConditions); }
+   * </pre>
+   *
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name to be created after copy.
+   * @param objectSource
+   *          Object name of the source.
+   * @param copyConditions
+   *          CopyConditions object with collection of supported CopyObject conditions.
+   *
+   * @throws InvalidBucketNameException
+   *           upon an invalid bucket name
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   */
+  public CopyObjectResult copyObject(String bucketName, String objectName, String objectSource,
```

Comment:
1. Returning CopyObjectResult is not useful. The method should return void if success and throws exception on failure.
2. Having `String srcBucketName, String srcObjectName` than `String objectSource` is better.  You would need to validate srcBucketName as well.